lib/src/tokenizer.dart

```diff
@@ -1869,7 +1869,7 @@ class HtmlTokenizer implements Iterator<Token> {
   }
 
   bool cdataSectionState() {
-    final data = <String>[];
+    final data = <String?>[];
```

Comment:
Almost certainly incorrect. The list is used for `join`, which means it should not contain any `null` values (they get into the string as `"null"`, which is never what you want).

Make this `<String>[];`.

I'm guessing the `if (ch == eof)` below is not recognized as comparing with `null`.
Consider changing `eof` to be declared as `const`. That might allow static analysis to recognize that you are comparing to `null` here.
Otherwise, change it to `if (ch == null)` and accept that `eof` isn't good enough post null-safety.

Reply:
Yeah, I already got rid of one alias for `null` in this package. I should probably get rid of `eof` too...

Switching this to `if (ch == null)` for now.

https://github.com/dart-lang/html/pull/136